---
title: "HTSeq Counts Analysis"
author: "Dave Bridges"
date: "August 2, 2015"
output:
  html_document:
    keep_md: true
---


```{r global_options, include=FALSE}
library(knitr)
opts_chunk$set(fig.path='figures/',
               echo=FALSE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
options(digits=0, scipen=8)
```

This script analyses the HTSeq resutls for the Juvenile HFD studies.  This script was most recently run on `r date()`

```{r data-entry}
input_folder <- '../server-scripts/htseq-output'
for (file in list.files(input_folder, full.names=T)) {
  # if the merged dataset doesn't exist, create it
  if (!exists("counts.table")){
    counts.table <- read.table(file, header=F, sep="\t", row.names='V1', col.names=c('V1',lapply(strsplit(file[1], "/"), '[[',4)))
  }
  if (exists("counts.table")){
  tmp <- read.table(file, sep='\t', 
                    row.names='V1', 
                    col.names=c('V1',lapply(strsplit(file[1], "/"), '[[',4)))
  counts.table <- cbind(counts.table, tmp)
  rm(tmp)
  }
}
counts.table <- counts.table[2:dim(counts.table)[2]]
```

This script merges all the files in the directory `r input_folder` and analyses the counts.

```{r htseq-counts-summary}
counts.table['Total',] <- colSums(counts.table)
#this is the sum of the unaligned rows at the bottom of the table
counts.table['Not-Aligned',] <- colSums(counts.table[grep("__", rownames(counts.table)),])
counts.table['Aligned',] <- counts.table['Total',] - counts.table['Not-Aligned',]
counts.table['Percent Aligned',] <- counts.table['Aligned',]/counts.table['Total',] * 100
counts.table['Percent Not In Gene',] <- counts.table['__no_feature',]/counts.table['Total',] * 100
counts.table['Percent Low Quality',] <- counts.table['__too_low_aQual',]/counts.table['Total',] * 100
library(RColorBrewer)
colorset <- brewer.pal(6, "Set2")
barplot(as.matrix(counts.table[c('Aligned','__no_feature','__too_low_aQual','__not_aligned','__ambiguous','__alignment_not_unique'),]), 
        las=2, col=colorset,cex.names=0.5,cex.axis=0.5,
        names.arg=lapply(strsplit(colnames(counts.table), ".counts"), '[[', 1))
legend("top", c("Mapped","Not in Gene","Low Quality","Not Aligned","Ambiguous","Not Unique"), bty="n", fil=colorset, cex = 0.5)
```

# Summary

On average `r mean(as.numeric(counts.table['Percent Aligned',]))`% of reads were aligned to a known feature, with `r mean(as.numeric(counts.table['Percent Not In Gene',]))`% of reads not mapped to known features.  A remaining `r mean(as.numeric(counts.table['Percent Low Quality',]))`% of reads were too low quality to be mapped.  The range of mapped reads was `r min(counts.table['Aligned',])` to `r max(counts.table['Aligned',])` with an average of `r mean(as.numeric(counts.table['Aligned',]))`.

# Annotation

```{r annotation}
annotation_file <- 'ENSEMBL Annotation.csv'
library(biomaRt)
if (!(annotation_file %in% dir())){
  annotation <- getBM(attributes=c('gene_biotype','external_gene_name', 'ensembl_gene_id'),
                    filters='ensembl_gene_id',
                    values=as.character(lapply(strsplit(rownames(counts.table), "\\."), "[[", 1)),
                    mart=useDataset('mmusculus_gene_ensembl',useMart('ensembl')))
  write.csv(annotation, annotation_file)
  }
```

# Session Information

```{r session-info}
sessionInfo()
```
